packages/element-app/src/components/DIDWallet/CreateWalletCard/CreateWalletDialog/CreateWalletDialog.js

```diff
@@ -39,25 +39,38 @@ class CreateWalletDialog extends React.Component {
     this.props.onClose();
   };
 
-  componentWillReceiveProps(nextProps) {
+  async componentWillReceiveProps(nextProps) {
     this.setState({
       open: nextProps.open,
       password: '',
       confirm: '',
       ciphered: '',
     });
-    const keys = ['#primary', '#recovery', '#edv'].map((kid) => {
-      const key = element.func.createKeys();
-      const walletKey = {
-        type: 'assymetric',
-        encoding: 'hex',
-        publicKey: key.publicKey,
-        privateKey: key.privateKey,
-        tags: ['Secp256k1VerificationKey2018', kid],
-        notes: '',
-      };
-      return walletKey;
-    });
+    const primaryKey = await element.crypto.secp256k1.createKeys();
+    const walletKey1 = {
+      type: 'assymetric',
+      encoding: 'hex',
+      publicKey: primaryKey.publicKey,
+      privateKey: primaryKey.privateKey,
+      tags: ['Secp256k1VerificationKey2018', '#primary'],
```

Comment:
Related issue: https://github.com/decentralized-identity/element/issues/132

basically, we are hacking the use of tags to find ed25519 keys in edv... until we have proper web kms support, its handy to generate tags that point to a fully qualified public key, for example:

`did:example:123#primary`... this is not easily achieved unless you generate the wallet file twice, first time to define keys, second time to update tags to use the did now that it is knowable.

without doing this, we will need to use the edv wallet editor before this will work out of the box.

Can you add some code here to generate a key specifically for edv use, that has a tag of the form:

`did:example:123#edv`

Reply:
Sorry this comment belongs on the Ed25519Key...